SoftLayer/managers/hardware.py

```diff
@@ -395,6 +395,24 @@ def get_create_options(self):
             'extras': extras,
         }
 
+    def get_order(self, order_id):
+        BILLING_ORDER_ENDPOINT = 'SoftLayer_Billing_Order'
+        BILLING_ORDER_MASK = 'mask[orderTopLevelItems[hostName,billingItem[id,provisionTransaction[hardware[id,hardwareStatus]]]]]'  # noqa
+
+        result = self.client[BILLING_ORDER_ENDPOINT].getObject(
+            id=order_id,
+            mask=BILLING_ORDER_MASK
+        )
+        return result
+
+    def get_hardware_id(self, order_id):
+        order = self.get_order(order_id)
+        return order\
+            ['orderTopLevelItems'][0]\
+            ['billingItem']\
+            ['provisionTransaction']\
+            ['hardwareId']
```

Comment:
So their whole sdk is just dicts upon dicts upon dicts? No business objects? The problem I have with this nested access is it's very easy for it to break - there's lots of keys that could potentially change if they change part of their sdk and we merge in a fix.